sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/HttpClientServiceVersionAugmentedArgumentsProvider.java

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.test.TestBase;
+import com.azure.core.test.TestMode;
+import com.azure.core.test.implementation.TestingHelpers;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.ServiceVersion;
+import com.azure.core.util.logging.ClientLogger;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+import org.junit.platform.commons.support.ReflectionSupport;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
```

Comment:
This changes the behavior of live, record a little no? Up until now the difference was simply one creaating a playbook and the other not, while with this the matrix that is getting tested also differs - no?

Reply:
Live testing will be allowed to run across multiple service versions with this change but playback and record will only be allowed to use the specified version or latest is unspecified.

Reply:
This can be a potential for confusion - I am assuming this is because we can have only one  record file per method. But - actually that was true with live tests with the current annotation too! so this will be in the similar vain.